File: apps/web/src/api/triage.ts
import client from './client'

export interface TriageRequest {
  symptoms_text: string
  age?: number
  sex?: string
  user_id?: number // Añadido para vincular la consulta al usuario
}

export interface AlternativeSpecialty {
  specialty_slug: string
  reason: string
}

export interface TriageResult {
  urgency: 'low' | 'medium' | 'high' | 'emergency'
  recommended_specialty_slug: string
  rationale_bullets: string[]
  clarifying_questions: string[]
  alternatives: AlternativeSpecialty[]
  red_flags_detected: string[]
  disclaimer: string
}

export interface TriageResponse {
  id: number
  result: TriageResult
  created_at: string
}

export const submitTriage = async (data: TriageRequest): Promise<TriageResponse> => {
  const response = await client.post<TriageResponse>('/ai/triage', data)
  return response.data
}

// Nueva función para obtener el historial
export const getTriageHistory = async (userId: number, limit: number = 3): Promise<TriageResponse[]> => {
  // Ajusta la ruta '/ai/triage/history' según cómo vayas a nombrar tu endpoint en el backend
  const response = await client.get<TriageResponse[]>('/ai/triage/history', {
    params: {
      user_id: userId,
      limit: limit
    }
  })
  return response.data
}

File: apps/api/app/schemas/triage.py
from pydantic import BaseModel, ConfigDict, Field
from typing import List, Optional
from datetime import datetime


class TriageRequest(BaseModel):
    symptoms_text: str = Field(..., min_length=10, description="Descripción de los síntomas")
    age: Optional[int] = Field(None, ge=0, le=150)
    sex: Optional[str] = Field(None, pattern="^(male|female|other)$")


class AlternativeSpecialty(BaseModel):
    specialty_slug: str
    reason: str


class TriageResult(BaseModel):
    urgency: str = Field(..., pattern="^(low|medium|high|emergency)$")
    recommended_specialty_slug: str
    rationale_bullets: List[str]
    clarifying_questions: List[str] = Field(default_factory=list, max_length=5)
    alternatives: List[AlternativeSpecialty] = Field(default_factory=list, max_length=3)
    red_flags_detected: List[str] = Field(default_factory=list)
    disclaimer: str


class TriageResponse(BaseModel):
    id: int
    result: TriageResult
    created_at: datetime
    
    model_config = ConfigDict(from_attributes=True)
